### ISSUE TITLE
Problem with pull-secretstream

### ISSUE THREAD

**Issue by staltz** [staltz/secret-handshake-ext#2]
1. Clone https://github.com/staltz/ppppp-sync
2. `npm it`
3. :heavy_check_mark: Tests pass
4. Update `secret-handshake-ext` from 0.0.8 to 0.0.9
5. `npm it`
6. :x: Tests fail

Can probably debug it with `DEBUG=* npm it`

**Comment by staltz** [comment]
cc @ahdinosaur

**Comment by ahdinosaur** [comment]
i had a thought this may be because of how libsodium padding works (which i spent some time yesterday learning): you need to leave an extra byte as a padding minimum, which i didn't do. can investigate later.

**Comment by ahdinosaur** [comment]
@staltz when i try to run `npm it` in `ppppp-sync`, i get:

```
  Error: db requires config.db.path
      at assertValidConfig (/home/dinosaur/repos/ahdinosaur/ppppp-sync/node_modules/ppppp-db/lib/index.js:84:11)
      at Object.initDB (/home/dinosaur/repos/ahdinosaur/ppppp-sync/node_modules/ppppp-db/lib/index.js:143:3)
      at /home/dinosaur/repos/ahdinosaur/ppppp-sync/node_modules/secret-stack/lib/api.js:52:28
      at Array.forEach (<anonymous>)
      at create (/home/dinosaur/repos/ahdinosaur/ppppp-sync/node_modules/secret-stack/lib/api.js:51:20)
      at createPeer (/home/dinosaur/repos/ahdinosaur/ppppp-sync/test/util.js:27:6)
      at TestContext.<anonymous> (/home/dinosaur/repos/ahdinosaur/ppppp-sync/test/account-sync.test.js:17:17)
      at Test.runInAsyncScope (node:async_hooks:203:9)
      at Test.run (node:internal/test_runner/test:571:25)
      at Test.start (node:internal/test_runner/test:484:17)
```

do i need a config file somewhere?

**Comment by staltz** [comment]
Oh, I just recently updated -db and forgot to update other modules. It should just be a matter of changing the config object in `/test/utils`

**Comment by ahdinosaur** [comment]
i see the error now. hmm...

fwiw, i don't see an error when using https://github.com/staltz/secret-handshake-ext/pull/3.

**Comment by ahdinosaur** [comment]
okay, i believe i've fixed the bug in `pull-secretstream@2.0.3`: https://github.com/ahdinosaur/pull-secretstream/commit/cc53f8139262b8d274faab821987e097a92b788a.

was indeed because there was no room for the minimum padding (1 byte) on full blocks.

**Comment by staltz** [comment]
Thanks! That seems like it. I updated shse in ppppp-sync and it's working. :)
